pkg/sources/sources.go

```diff
@@ -223,3 +243,21 @@ func (p *Progress) GetProgress() *Progress {
 	defer p.mut.Unlock()
 	return p
 }
+
+// CommonEnumerationOk is a helper function to construct an EnumerationResult
+// using a CommonSourceUnit.
+func CommonEnumerationOk(id string) EnumerationResult {
+	return EnumerationResult{
```

Comment:
nit: We can omit the zero value for Error if you want. You can then inline the thing.
`return EnumerationResult{Unit: CommonSourceUnit{ID: id}}`

Leaving it is also fine if we feel like it adds extra clarity. I'm leaning toward removing it only b/c your comment for `ENumerationResult` mentions the fields being mutually exclusive. Similar comment for EumerationErr.